ext/openssl/openssl.c

```diff
@@ -1496,15 +1609,15 @@ PHP_FUNCTION(openssl_spki_new)
 	if (spki != NULL) {
 		NETSCAPE_SPKI_free(spki);
 	}
-	if (keyresource == NULL && pkey != NULL) {
+	if (free_pkey && pkey != NULL) {
 		EVP_PKEY_free(pkey);
 	}
 
 	if (s && ZSTR_LEN(s) <= 0) {
 		RETVAL_FALSE;
 	}
 
-	if (keyresource == NULL && s != NULL) {
+	if (free_pkey && s != NULL) {
```

Comment:
This doesn't make sense, `s` and `free_pkey` have no relation. This branch should just be dropped. (Because free_pkey is always false in this code, it ends up not mattering.)